AppDB/appscale/datastore/datastore_distributed.py

```diff
@@ -1002,10 +1003,21 @@ def __get_start_key(self, prefix, prop_name, order, last_result):
       plist = ent.property_list()
 
     val = None
-    for p in plist:
-      if p.name() == prop_name:
-        val = str(encode_index_pb(p.value()))
-        break
+
+    # If the entity provided by the cursor has multiple values for the given
+    # property and the query has information about what the value should be,
+    # use the value from the query.
```

Comment:
Wouldn't be more accurate to say:
> Use the value from the query if possible.

This if statement doesn't actually check if entity has multiple values for the given property.